v2/kafka-to-pubsub/src/main/java/com/google/cloud/teleport/v2/templates/KafkaToPubsub.java

```diff
@@ -0,0 +1,177 @@
+ * servers are specified by the user as template parameters.
+ *
+ * <p><b>Pipeline Requirements</b>
+ *
+ * <ul>
+ *   <li>Kafka Bootstrap Server(s).
+ *   <li>Kafka Topic(s) exists.
+ *   <li>The PubSub output topic exists.
+ * </ul>
+ *
+ * <p><b>Example Usage</b>
+ *
+ * <pre>
+ * # Set the pipeline vars
+ * PROJECT=id-of-my-project
+ * BUCKET_NAME=my-bucket
+ *
+ * # Set containerization vars
+ * IMAGE_NAME=my-image-name
+ * TARGET_GCR_IMAGE=gcr.io/${PROJECT}/${IMAGE_NAME}
+ * BASE_CONTAINER_IMAGE=my-base-container-image
+ * BASE_CONTAINER_IMAGE_VERSION=my-base-container-image-version
+ * TEMPLATE_PATH="gs://${BUCKET_NAME}/templates/kafka-pubsub.json"
+ * TARGET_GCR_IMAGE=gcr.io/${PROJECT}/${IMAGE_NAME}
+ *
+ * # Create bucket in the cloud storage
+ * gsutil mb gs://${BUCKET_NAME}
+ *
+ * # Go to the beam folder
+ * cd /path/to/DataflowTemplates/v2
+ *
+ * <b>FLEX TEMPLATE</b>
+ * # Assemble jar with dependencies
+ * mvn package -am -pl kafka-to-pubsu
+ *
+ * # Go to the template folder
+ * cd /path/to/DataflowTemplates/v2/kafka-to-pubsub
+ *
+ * # Build the flex template
+ * gcloud dataflow flex-template build ${TEMPLATE_PATH} \
+ *       --image-gcr-path "${TARGET_GCR_IMAGE}" \
+ *       --sdk-language "JAVA" \
+ *       --flex-template-base-image ${BASE_CONTAINER_IMAGE} \
+ *       --metadata-file "src/main/resources/kafka_to_pubsub_metadata.json" \
+ *       --jar "cd /path/to/DataflowTemplates/v2/kafka-to-pubsub" \
+ *       --env FLEX_TEMPLATE_JAVA_MAIN_CLASS="org.apache.beam.templates.KafkaToPubsub"
+ *
+ * # Execute template:
+ * API_ROOT_URL="https://dataflow.googleapis.com"
+ * TEMPLATES_LAUNCH_API="${API_ROOT_URL}/v1b3/projects/${PROJECT}/templates:launch"
+ * JOB_NAME="kafka-to-pubsub-`date +%Y%m%d-%H%M%S-%N`"
+ *
+ * time curl -X POST -H "Content-Type: application/json"     \
+ *     -H "Authorization: Bearer $(gcloud auth print-access-token)" \
+ *     "${TEMPLATES_LAUNCH_API}"`
+ *     `"?validateOnly=false"`
+ *     `"&dynamicTemplate.gcsPath=${BUCKET_NAME}/path/to/image-spec"`
+ *     `"&dynamicTemplate.stagingLocation=${BUCKET_NAME}/staging" \
+ *     -d '
+ *      {
+ *       "jobName":"${JOB_NAME}",
+ *       "parameters": {
+ *           "bootstrapServers":"broker_1:9092,broker_2:9092",
+ *           "inputTopics":"topic1,topic2",
+ *           "outputTopic":"projects/${PROJECT}/topics/your-topic-name"
+ *        }
+ *       }
+ *      '
+ * </pre>
+ */
+public class KafkaToPubsub {
+
+  /* Logger for class.*/
+  private static final Logger LOG = LoggerFactory.getLogger(KafkaToPubsub.class);
+
+  /**
+   * Main entry point for pipeline execution.
+   *
+   * @param args Command line arguments to the pipeline.
+   */
+  public static void main(String[] args) {
+    KafkaToPubsubOptions options =
+        PipelineOptionsFactory.fromArgs(args).withValidation().as(KafkaToPubsubOptions.class);
+
+    run(options);
+  }
+
+  /**
+   * Runs a pipeline which reads message from Kafka and writes it to GCS.
```

Comment:
Runs a pipeline which reads message from Kafka and writes to Pub/Sub

Reply:
fixed